### PR TITLE
Submit group 1's week 8 assignment

### PR DESCRIPTION
**Pull request by sbrown5x**
Hi all! We will push files as they are ready. Looking forward to your feedback! I'll let my fellow group members add anything they'd like feedback on in the other files, but for the `simulate.do` file, I'd be curious for your thoughts on 1) if there's a more elegant way to make the unique ID variable and 2) if there's a different way we should be creating/conceptualizing the strata in the data. Thank you!

### CODE REVIEW

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:17]: Why a chi-squared distribution? An explanation would be helpful.

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:21]: Where do these numbers (e.g., 0.3) come from? Explanation would be helpful.

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:30]: Am I correctly understanding that universities either have exactly 500 (30%), exactly 750 (50%), or exactly 1,000 students (20%)? If so, is there a reason you are using such precise numbers?

If I understand this correctly, perhaps consider stratifying using a more-nuanced procedure. First, you could assign each university a random number called temp. Then, sort the data by temp. Then, say, for universities with 0<temp<0.3, generate students from, say, runinformint(500,750). Repeat for other intervals of temp. This approach, I believe, could be applied to many of your DGPs.

Please let me know if I have been unclear or mistaken in my understanding, or if you have concerns about my recommended approach.

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:42]: FYI, you can just write "rnormal(1200,70)" instead of "1200+rnormal(0,70)". The first parameter specifies the mean of the distribution.

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:43]: What does this do?

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:47]: Why a Poisson distribution?

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:49]: Why are you top-censoring at 8 APs? AFAIK, there is no maximum (e.g., I took 12).

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [inline _Week8/group1/simulate.do:58]: Where does 34 majors come from?

**Review by noahblakesmith — CHANGES_REQUESTED** [review]
Very nice job, group 1. Per @sbrown5x, I recommended a possible approach to stratification that may be more elegant. Looking forward to reviewing the README once submitted!

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:17]: Good question--we'll make a note to explain in the README file!

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:21]: See my response to your comment below.

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:30]: There was no reason for the precise numbers other than we wanted to randomize it a bit, so they were arbitrarily selected. Good idea to randomize it even further though, I'll make a note to edit. Could you clarify what you mean by DGPs?

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:42]: Got it, thank you!!

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:43]: The trunc command rounds it to an integer!

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:47]: Good question--not sure we had a justification other than knowing that we wanted 1) it to be an integer and 2) to vary the kinds of distributions we used. We can maybe see if there's a distribution of AP classes available online, I'll save this comment so that we remember to edit accordingly.

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:49]: Honestly, I didn't finish high school in the US, so this was my guesstimate as to what was usually on the higher end! But I can look into it more thoroughly and change the number.

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:58]: I had trouble finding a number online so put a guesstimate. Leaving a note to at least replace it with the # of majors that Georgetown offers (knowing that it'll vary a lot by university).

File: _Week8/group1/simulate.do
Comment (noahblakesmith) [reply _Week8/group1/simulate.do:30]: Got it. DGP = data generating process. I meant that you could apply the procedure I explained to a few of the instances in which you randomly generate data. Please let me know if you have further questions/clarifications.

File: _Week8/group1/simulate.do
Comment (bbdaniels) [reply _Week8/group1/simulate.do:43]: Check the documentation on this -- `trunc()` does not in fact round, rather it discards everything after the decimal place! That won't matter enormously here, but in general you will want to prefer `round()` so that numbers with decimal parts > 0.5 become the _next_ integer.

File: _Week8/group1/simulate.do
Comment (bbdaniels) [reply _Week8/group1/simulate.do:49]: That is fine! But if you do want to limit it, consider whether you want the distribution to be effectively top-coded (ie, you are overweighting 8 slightly by binning all results above 8 into 8). You might, you might not!

**Comment by bbdaniels** [comment]
Did I mess something up here and remove the documentation? If so you should have a copy on your local -- let me know so I can fix that 😬 (if not please change to draft PR until you are ready so I don't worry about it)

File: _Week8/group1/simulate.do
Comment (bbdaniels) [reply _Week8/group1/simulate.do:30]: Note that `expand` can take a variable as an argument instead of a fixed value, which can be very useful! That is, by using `trunc` or `round` as you have in other places you can very quickly create a distribution of sizes!

Comment (sbrown5x) [reply _Week8/group1/simulate.do:30]: @f2pHgty8hw: Yes, I realized what you meant during class today! Thanks.

@bbdaniels: Oh got it! I was wondering how we could have used `expand` here... during editing, I'll see where we can use that command.

File: _Week8/group1/simulate.do
Comment (sbrown5x) [reply _Week8/group1/simulate.do:43]: @bbdaniels : Ah, thank you! I should have read that more closely (and realized of course the command would be called `round`, because truly what else would it be called...)

**Comment by sbrown5x** [comment]
> Did I mess something up here and remove the documentation? If so you should have a copy on your local -- let me know so I can fix that 😬 (if not please change to draft PR until you are ready so I don't worry about it)

No worries, nothing was messed up! A couple of group members needed extra time this week and so have not pushed yet. Wanted to add files as we had them so we didn't hold up the peer reviewers but we can keep the PR as a draft in the future.

**Comment by sbrown5x** [comment]
[Just adding a note that I know we should make the changes mentioned in the feedback, but that I'm going to hold off until all of our files have been submitted!]

**Comment by sbrown5x** [comment]
Hi @f2pHgty8hw @MiglePetrou @Benjamin-Tu! Just wanted to let you know all of our files have been uploaded and are ready for your review. Apologies for the delay!!

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:13]: Does the total number of observations stay fixed as the number of strata changes? Be precise if your description of the data. Introduce state-level fixed effects in the prior paragraph? What does "uniformly randomize" mean? Can you describe the parameters more specifically?

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:15]: Be clear about the relative scales and importance of the university vs state fixed effects. What is the function form of the GPA variable? What is the "true coefficient on GPA" -- is this the true increase in expected salary per grade point? What is the total range of possible salaries you will generate? Is $5,000 large or small relative to the total variation?

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:17]: What variables and why?

For ID variables -- don't use "leading zero" approaches. Excel will mangle these hopelessly. I generally recommend strings (by, for example, adding a leading letter to this approach instead of a zero).

Comment (bbdaniels) [reply _Week8/group1/README.md:17]: Second -- you may or may not _want_ IDs to be identifying in this way, if you need to publish data.

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:19]:
```suggestion
Lastly, we run the regression of salary on GPA, undergraduate major, state in which their university is located, and university attended, in order to estimate the coefficient on GPA. The program ends by returning the beta coefficient and standard error on GPA, as well as the total sample size. 
```

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:24]: This approach seems a little unclear to me, both in function and purpose. Why are you varying the number of states? Why only draw 5 samples?

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:28]: Don't present full data -- present a summary by using `collapse` by the parameters of interest for your simulation!

File: _Week8/group1/README.md
Comment (bbdaniels) [inline _Week8/group1/README.md:30]: The second one here seems odd -- you are illustrating a relationship, not a distribution, so saying it "appears normal" does not have any meaning here.

For the third -- remember -- these are _strata_ as you correctly explained earlier, not _clusters_. So are you not just showing that standard errors decrease with N here, since your strata mechanically increase the sample size? Imagine an alternative approach -- where you hold the total N fixed but distribute them across more strata. How would _that_ affect your precision?

**Review by bbdaniels — APPROVED** [review]
Great work in writing up the programs and the explanation. For the final project, you will want to slightly revise this so that it aligns more closely in _concept_ with your project proposal. I think some of the motivation for the exact simulation here is not quite lined up the way. you would want to discuss the statistical power of your proposed project, but it is very much in the right direction! Approving now and I will merge in to solve the conflict.

**Pull request merged**
